src/components/DeleteGraphDialog.vue

```diff
@@ -99,14 +110,7 @@ export default Vue.extend({
   watch: {
     dialog() {
       if (this.dialog) {
-        this.timeout = window.setTimeout(() => {
-          this.disabled = false;
-          this.timeout = undefined;
-        }, 2000);
-      } else {
-        window.clearTimeout(this.timeout);
-        this.disabled = true;
-        this.timeout = undefined;
+        this.confirmationPhrase = randomPhrase();
       }
     },
   },
```

Comment:
Also should mention I tested my suggestion locally with no issues

Reply:
This is a good thought, though I think technically we're not supposed to use a computed prop here. I have an intuition that computed props are supposed to be pure functions of their dependencies, which this decidedly is not (because of the embedded call to `Math.random()`). I've experimented with computed props that have side effects before, and it didn't go well. To be fair, the side effect in this case is pretty light, and this might have the right semantics (only update the confirmation phrase when the dialog is toggled). Even so, it makes me uncomfortable.

However, if @subdavis thinks this is a safe place to use a computed prop, I will change it.

Reply:
I would actually argue that this doesn't have any side effects. While it's not a mathematically pure function (due to the random call), this doesn't modify any other part of the state, and evaluates when `this.dialog` changes. To me, having sole assignments to component state inside of watchers sets of alarm bells, as watchers are usually only used when needed, and this functionality is completely captured in a computed prop.

That being said, I'd also be interested in @subdavis's opinion.

Reply:
Because of how computed prop caching works (And having not looked at the rest of the code) I believe the computed property should work as long as you're verifying the computed phrase while the dialog is open (thereby receiving the cached value).

However, I agree with @waxlamp that using a watcher is better style here.  Having to think about computed cache behavior at all while working on a feature like this I think is a smell.  Also, when the next less vue-savvy developer needs the value after the dialog closed, I could see this being [an hours-long gotcha](https://en.wikipedia.org/wiki/Principle_of_least_astonishment).

  I vote watcher.

Reply:
Thanks for this analysis, @subdavis.

I am caught between equally good arguments on this, though the principle of least astonishment is pushing me towards leaving this mediated through a watcher. The important bit is, even though this has the right semantics *for our use case*, in general it violates the idea of what a computed property is: a value that depends *on the values of the dependencies*. That is, technically, without knowing anything else about this, one would expect for that computed prop to have at most two values: one when the dialog is open, and one when it's closed. Even though it feels like a clever shortcut, this seems like one of those cases where explicit is better than implicity.

@AlmightyYakob, let me know if you have any major concerns left about this. We can always revisit later if the watcher style causes unforeseen problems.

Reply:
I can see your guys arguments about this, so I'm okay leaving it.

Reply:
Thanks for the open discussion, everyone!